src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/edit-video-unit/edit-video-unit.component.ts

```diff
@@ -0,0 +1,73 @@
+import { Component, OnInit } from '@angular/core';
+import { VideoUnit } from 'app/entities/lecture-unit/videoUnit.model';
+import { VideoUnitFormData } from 'app/lecture/lecture-unit/lecture-unit-management/video-unit-form/video-unit-form.component';
+import { ActivatedRoute, Router } from '@angular/router';
+import { VideoUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/videoUnit.service';
+import { JhiAlertService } from 'ng-jhipster';
+import { onError } from 'app/shared/util/global.utils';
+import { finalize, switchMap, take } from 'rxjs/operators';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+
+@Component({
+    selector: 'jhi-edit-video-unit',
+    templateUrl: './edit-video-unit.component.html',
+    styles: [],
+})
+export class EditVideoUnitComponent implements OnInit {
+    isLoading = false;
+    videoUnit: VideoUnit;
+    formData: VideoUnitFormData;
+    lectureId: number;
+
+    constructor(private activatedRoute: ActivatedRoute, private router: Router, private videoUnitService: VideoUnitService, private alertService: JhiAlertService) {}
+
+    ngOnInit(): void {
+        this.isLoading = true;
+        this.activatedRoute.paramMap
+            .pipe(
+                take(1),
+                switchMap((params) => {
+                    const videoUnitId = Number(params.get('videoUnitId'));
+                    this.lectureId = Number(params.get('lectureId'));
+                    return this.videoUnitService.findById(videoUnitId);
+                }),
+                finalize(() => {
+                    this.isLoading = false;
+                }),
+            )
+            .subscribe(
+                (videoUnitResponse: HttpResponse<VideoUnit>) => {
+                    this.videoUnit = videoUnitResponse.body!;
```

Comment:
can this never be null?  Maybe it's safer to assign the values via  
```
name: videoUnitResponse.body?.name
```
..etc.

Reply:
The body should never be null. If the corresponding video unit is not found, a httpError will be thrown.